### ISSUE TITLE
annotation of annotation array is not readable in .class files

### ISSUE THREAD

**Issue by yigit** [google/ksp#356]
For the given code:

```
            @MainAnnotation(
                typeList = [String::class, Int::class],
                singleType = Long::class,
                intMethod = 3,
                otherAnnotationArray = [
                    OtherAnnotation(
                        value = "other list 1"
                    ),
                    OtherAnnotation(
                        value = "other list 2"
                    )
                ],
                singleOtherAnnotation = OtherAnnotation("other single")
            )
            public class Subject
```

When I read `otherAnnotationArray` from `MainAnnotation`, it only works if it is in source but does not work if it is in `.class` file. The retention is `RUNTIME` in both annotations and also working fine with KAPT.

When debugging, I see the `KSValueArgumentLiteImpl` has an empty array value.

![image](https://user-images.githubusercontent.com/89202/111238889-4d74b100-85b5-11eb-85c1-245912c6aa2d.png)

**Comment by yigit** [comment]
oh this is scary...
I reproduced it and the annotation descriptor literally has 0 element array for its value :/

https://github.com/yigit/ksp/commit/ad7c197de5c07bd7155230eb2d0c1e222ebab52b

**Comment by ting-yuan** [comment]
Kotlin compiler didn't implement this. Fortunately, the annotations come directly from bytecode rather than metadata. Therefore once the compiler implements it, this issue is automatically fixed. The PR to compiler is under review: https://github.com/JetBrains/kotlin/pull/4258

**Comment by ting-yuan** [comment]
Fix is now available in Kotlin 1.5.10.
